wandb/sdk/wandb_init.py

```diff
@@ -752,6 +752,11 @@ def init(self) -> Union[Run, RunDisabled, None]:  # noqa: C901
         # as the run is not upserted
         for k, v in self.init_artifact_config.items():
             run.config.update({k: v}, allow_val_change=True)
+        job_artifact = run._launch_artifact_mapping.get(
+            wandb.util.LAUNCH_JOB_ARTIFACT_SLOT_NAME
+        )
+        if job_artifact:
+            run.use_artifact(job_artifact)
```

Comment:
Rather than having this be a special case of the existing WANDB_ARTIFACT s input, should I make a new env var to express to a run it is sourced from a job?

Reply:
If a job is always gonna be tied to artifacts that seems fine. How many other things are in the `WANDB_ARTIFACTS` dict?

Reply:
It comes from the now hidden artifacts key in the overrides section. Before we'd allow users to overwrite input artifacts directly, but with jobs were saying that all inputs should be through the config.